### ISSUE TITLE
"Type" event can't be editable

### ISSUE THREAD

**Issue by mrcrony** [IFB-ElixirFr/ifbcat#111]
Hi,

We can't add type events in the admin.
"Under the wood" question here:

Is a covid-19 actions could be related to events ? (and adding "other" as event types, quick fix, seems ok to me to have particular events type in the futur)
Or
Could a project wouldn't have a team associated ? (sounds the wrong way to me) 

<img width="422" alt="Capture d’écran 2020-09-11 à 14 27 06" src="https://user-images.githubusercontent.com/8360521/92930389-4d4afb80-f442-11ea-8c54-5e21dc52332c.png">

**Comment by bryan-brancotte** [comment]
Note that event's type are [hard coded](https://github.com/IFB-ElixirFr/ifbcat/blob/d944ae33d15b92c2d7b89f80c084e27c534d6083/ifbcatsandbox_api/model/event.py#L88), but could easily move to mimics [EventCost](https://github.com/IFB-ElixirFr/ifbcat/blob/d944ae33d15b92c2d7b89f80c084e27c534d6083/ifbcatsandbox_api/model/event.py#L32-L45)

**Comment by joncison** [comment]
Event->type is a controlled vocabulary, so it's not supposed to be changed - at least not by typical end-users.  This goes for the other enums too  (see [here ](https://docs.google.com/spreadsheets/d/1tMzwkZINvFTj5mUOCBA8MrR8kxjrD6GRfyqH_mT1pi4/edit#gid=346104586)and [here](https://docs.google.com/spreadsheets/d/1tMzwkZINvFTj5mUOCBA8MrR8kxjrD6GRfyqH_mT1pi4/edit#gid=1411184705)).  It'd be good to know what "Other" events are - perhaps the enum could be extended?

Best place to annotate "COVID-19" relevant tools is using Event->keywords.

The Project::Team relationship could have been put on the side of Project but wasn't, mostly because BioinformaticsTeam is probably the most important entity, so things got parked there by default.

**Comment by mrcrony** [comment]
I like the EventType editable like EventCost...
But I don't like that the fact of deleting an EventCost attribute is allowed even if in use...

Can the relationships checking can be enforced simply ?

I understand that it will be a major issue with global (and partly...) imports procedures...
So a workaround, to avoid advanced useless user ACL might be to have a layer for the entire relationships "opened" on import, and then a layer while import is done ?
Might sound crazy, but a query that change the relationships on the fly (while importing) a sort of store procedure on import ? (don't kill me...)

**Comment by joncison** [comment]
hi @bryan-brancotte and @mrcrony

If you tried to delete an EventCost (via the Django admin interface) you would get a warning that to do so, would also delete all references to that object.  Given only superuser will have access to this interface, I think this is OK?

As for EventType being editable (via Django admin interface), I agree that'd be good - but then we should probably (to be consistent) do that for all the controlled vocabularies, i.e. things that are currently hard-coded through use of ``models.TextChoices``

**Comment by mrcrony** [comment]
Agree on both points:
- this is crucial that only super users got the right to delete those attributes. Side effects would be quite inconvenient...
- I temper that by only doing so while the need is here (meaning regular changes). Alternatively, a "how to" to clarify to super admins how to do so into the code / docker properly ? (I guess @hmenager  and @thomasrosnet would handle these cases in the futur ? Might come at some point also)

**Comment by thomasrosnet** [comment]
I agree with the second point of Nicolas, we should establish a Developper Manual or Maintenance Guide. 
This could be usefull to indicate the main things to know on where and what to edit in the code the interact with a given feature. So anyone who came later in the project will be able to quickly understand it and work on it (i also think of the future CDD that will work on it).
This doesn't need to go too much into details, only the main aspects to keep it short enough and clear, like a quick start guide.
